src/library/scala/math/BigInt.scala

```diff
@@ -382,7 +581,9 @@ final class BigInt(val bigInteger: BigInteger)
    *  it will be converted to `Double.NEGATIVE_INFINITY` or
    *  `Double.POSITIVE_INFINITY` as appropriate.
    */
-  def doubleValue: Double = this.bigInteger.doubleValue
+  def doubleValue: Double =
+    if (isValidLong && (-(1L << 53) <= _long && _long <= (1L << 53))) _long.toDouble
```

Comment:
is it worth reading `_long` into a local variable so it only gets read once, like you did in `/%`?

Reply:
The field is not volatile; I assume this will be optimized by the JVM. Note that /% calls longEncoding (as a lot of other methods), and we'll encounter the same problem there. I'd err in the direction of readability when in doubt.